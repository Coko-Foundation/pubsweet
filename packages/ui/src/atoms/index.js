export { default as Colorize } from './Colorize'
export { default as UploadingFile } from './UploadingFile'
export { default as Icon } from './Icon'
export { default as Attachment } from './Attachment'
export { default as Avatar } from './Avatar'
export { default as Badge } from './Badge'
export { default as Button } from './Button'
export { default as Checkbox } from './Checkbox'
export { default as File } from './File'
export { default as Menu } from './Menu'
export { default as Radio } from './Radio'
export { default as Tags } from './Tags'
export { default as Spinner } from './Spinner'
export { default as TextField } from './TextField'
export { default as ValidatedField } from './ValidatedField'
export { default as StateItem } from './StateItem'
export { default as AlignmentBox } from './AlignmentBox'
export { default as UploadButton } from './UploadButton'
export { default as Flexbox } from './Flexbox'
export { default as Link } from './Link'
export { default as Heading, H1, H2, H3, H4, H5, H6 } from './Heading'
export { default as ErrorText } from './ErrorText'
export { default as CenteredColumn } from './CenteredColumn'
export { default as Section } from './Section'
export { default as FilePicker } from './FilePicker'
