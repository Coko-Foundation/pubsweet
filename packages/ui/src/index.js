/* atoms */
export { default as Attachment } from './atoms/Attachment'
export { default as Avatar } from './atoms/Avatar'
export { default as Badge } from './atoms/Badge'
export { default as Button } from './atoms/Button'
export { default as Checkbox } from './atoms/Checkbox'
export { default as File } from './atoms/File'
export { default as Icon } from './atoms/Icon'
export { default as Menu } from './atoms/Menu'
export { default as Radio } from './atoms/Radio'
export { default as Tags } from './atoms/Tags'
export { default as TextField } from './atoms/TextField'
export { default as ValidatedField } from './atoms/ValidatedField'
export { default as StateItem } from './atoms/StateItem'
export { default as AlignmentBox } from './atoms/AlignmentBox'
export { default as UploadButton } from './atoms/UploadButton'
export { default as ControlGroup } from './atoms/ControlGroup'
export { default as Link } from './atoms/Link'
export { default as Title } from './atoms/Title'
export { default as ErrorText } from './atoms/ErrorText'
export { default as CenteredColumn } from './atoms/CenteredColumn'

/* molecules */
export { default as AppBar } from './molecules/AppBar'
export { default as Attachments } from './molecules/Attachments'
export { default as CheckboxGroup } from './molecules/CheckboxGroup'
export { default as EditableValue } from './molecules/EditableValue'
export { default as Files } from './molecules/Files'
export { default as FileUploadList } from './molecules/FileUploadList'
export { default as PlainButton } from './molecules/PlainButton'
export { default as Supplementary } from './molecules/Supplementary'
export { default as RadioGroup } from './molecules/RadioGroup'
export { default as YesOrNo } from './molecules/YesOrNo'
export { default as StateList } from './molecules/StateList'
export {
  default as AlignmentBoxWithLabel,
} from './molecules/AlignmentBoxWithLabel'
export { default as AlignmentTool } from './molecules/AlignmentTool.js'
export { default as Steps } from './molecules/Steps'

/* helpers */
export { default as theme } from './helpers/theme'
