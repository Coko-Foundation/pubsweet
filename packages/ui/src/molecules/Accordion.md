A component that can be expanded or collapsed.

```js
<Accordion label="Header" startExpanded>
  <hr />
  <span>the possibilities</span>
  <hr />
  <span>are endless</span>
  <hr />
  <Accordion label="Inception">
    <hr />
    <span>He's a good kid and a devil behind the wheel.</span>
    <hr />
  </Accordion>
</Accordion>
```

* The children are responsable with styling inside the accordion.

```js
<Accordion label="Big lorem ipsum">
  <div style={{ margin: 16 }}>
    What is Lorem Ipsum? Lorem Ipsum is simply dummy text of the printing and
    typesetting industry. Lorem Ipsum has been the industry's standard dummy
    text ever since the 1500s, when an unknown printer took a galley of type and
    scrambled it to make a type specimen book. It has survived not only five
    centuries, but also the leap into electronic typesetting, remaining
    essentially unchanged. It was popularised in the 1960s with the release of
    Letraset sheets containing Lorem Ipsum passages, and more recently with
    desktop publishing software like Aldus PageMaker including versions of Lorem
    Ipsum. Why do we use it? It is a long established fact that a reader will be
    distracted by the readable content of a page when looking at its layout. The
    point of using Lorem Ipsum is that it has a more-or-less normal distribution
    of letters, as opposed to using 'Content here, content here', making it look
    like readable English. Many desktop publishing packages and web page editors
    now use Lorem Ipsum as their default model text, and a search for 'lorem
    ipsum' will uncover many web sites still in their infancy. Various versions
    have evolved over the years, sometimes by accident, sometimes on purpose
    (injected humour and the like).
  </div>
</Accordion>
```

* Change header icon

```js
<Accordion
  data-test-id="test-me"
  label="My icon is different"
  icon="arrow-down"
>
  <hr />
  <span>it's ok to be different</span>
  <hr />
</Accordion>
```

* Custom header

```js
const MyHeader = ({ toggle }) => {
  return <div onClick={toggle} style={{
    backgroundColor: 'lavender',
    display: 'flex',
    justifyContent: 'space-between',
  }}>
    <span>CUSTOM HEADER SO NICE</span>
    <span>Some info here</span>
  </div>
}

<Accordion header={MyHeader}>
 <hr />
  <span>thanos stars</span>
  <hr />
  <span>in deadpool 2</span>
  <hr />
</Accordion>
```
