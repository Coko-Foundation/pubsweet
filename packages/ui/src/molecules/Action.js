/*
  Actions arose from current designs (like the Appbar) where we had to blend
  links and buttons, but make them appear the same.

  The Action centralizes the visual part of this scenario, while leaving the
  underlying mechanics of links and buttons intact.

  -- TODO
  THIS COULD BE REMOVED IN THE FUTURE, AS IT IS UNCLEAR WHETHER WE SHOULD
  HAVE LINKS AND BUTTONS THAT LOOK THE SAME IN THE FIRST PLACE.
*/

import React from 'react'
import { css } from 'styled-components'
import { th } from '@pubsweet/ui-toolkit'

import OriginalButton from '../atoms/Button'
import OriginalLink from '../atoms/Link'

const common = css`
  color: ${th('colorPrimary')};
  font: ${th('fontInterface')};
  font-size: ${th('fontSizeBase')};
  font-weight: ${props => (props.active ? 'bold' : 'normal')};
  text-decoration: none;
  transition: ${th('transitionDuration')} ${th('transitionTimingFunction')};

  &:hover,
  &:focus,
  &:active {
    background: none;
    color: ${th('colorPrimary')};
    text-decoration: underline;
  }

  ${th('cssOverrides.Action')};
`

const Button = OriginalButton.extend`
  background: none;
  border: none;
  min-width: 0;
  padding: 0;

  ${common};
`

const Link = OriginalLink.extend`
  ${common};
`

const Action = props => {
  if (props.to) return <Link {...props}>{props.children}</Link>
  return <Button {...props}>{props.children}</Button>
}

export default Action
