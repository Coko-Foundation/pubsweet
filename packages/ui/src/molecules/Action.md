Actions arose from current designs (like the Appbar) where we had to blend
links and buttons, but make them appear the same.  
The Action centralizes the visual part of this scenario, while leaving the
underlying mechanics of links and buttons intact.

Action that is a link.

```js
<Action to="/">I am a link.</Action>
```

Action that is a button.

```js
<Action>I am a button.</Action>
```

An action can be marked as active. (useful for a navigation bar)

```js
<Action active>Active</Action>
```
