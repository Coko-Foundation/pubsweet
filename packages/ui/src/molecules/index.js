export { default as Accordion } from './Accordion'
export { default as Action } from './Action'
export { default as ActionGroup } from './ActionGroup'
export { default as AlignmentBoxWithLabel } from './AlignmentBoxWithLabel'
export { default as AlignmentTool } from './AlignmentTool'
export { default as AppBar } from './AppBar'
export { default as Attachments } from './Attachments'
export { default as CheckboxGroup } from './CheckboxGroup'
export { default as EditableValue } from './EditableValue'
export { default as Files } from './Files'
export { default as FileUploadList } from './FileUploadList'
export { default as Menu } from './Menu'
export { default as PlainButton } from './PlainButton'
export { default as RadioGroup } from './RadioGroup'
export { default as StateList } from './StateList'
export { default as Steps } from './Steps'
export { default as Supplementary } from './Supplementary'
export { default as Upload } from './Upload'
export { default as YesOrNo } from './YesOrNo'
export { default as Tabs } from './Tabs'
