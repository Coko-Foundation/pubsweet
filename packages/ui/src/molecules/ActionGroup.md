ActionGroup is a list of Actions displayed inline

```js
import { Action, ActionGroup } from '@pubsweet/ui'
;<ActionGroup>
  <Action to="/edit">Edit</Action>
  <Action>Rename</Action>
  <Action>Delete</Action>
</ActionGroup>
```
