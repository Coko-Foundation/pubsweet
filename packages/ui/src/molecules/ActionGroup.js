import React from 'react'
import styled from 'styled-components'
import { th } from '@pubsweet/ui-toolkit'

import Action from './Action'

const Group = styled.div`
  ${th('cssOverrides.ActionGroup')};
  ${th('cssOverrides.ActionGroup.Root')};
`

const ActionWrapper = styled.div`
  display: inline-block;
  padding: 0 ${th('subGridUnit')};
  ${th('cssOverrides.ActionGroup.ActionWrapper')};
`

const ActionGroup = props => {
  const children = React.Children.map(props.children, child => (
    <ActionWrapper>{child}</ActionWrapper>
  ))

  return <Group>{children}</Group>
}

ActionGroup.propTypes = {
  children: (props, propName, componentName) => {
    const prop = props[propName]
    let error = null

    React.Children.forEach(prop, child => {
      if (child.type !== Action) {
        error = new Error('ActionGroup should only take Actions as children')
      }
    })

    return error
  },
}

export default ActionGroup
