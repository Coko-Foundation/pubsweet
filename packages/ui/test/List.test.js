import React from 'react'
import { shallow } from 'enzyme'

import { List } from '../src/molecules'

describe('<List>', () => {
  it('renders', () => {
    const wrapper = shallow(<List />)
    expect(wrapper).toMatchSnapshot()
  })

  it('displays all the items', () => {
    const items = [
      { id: 1, name: 'First name' },
      { id: 2, name: 'Second name' },
    ]
    const wrapper = shallow(<List items={items} />)
    expect(wrapper.children()).toHaveLength(2)
    expect(wrapper).toMatchSnapshot()
  })
})
