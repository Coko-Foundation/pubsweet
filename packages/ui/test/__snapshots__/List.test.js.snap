// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List> renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <List />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [],
    },
    "ref": null,
    "rendered": Array [],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [
        Object {
          "data-test-id": [Function],
        },
      ],
      "componentStyle": ComponentStyle {
        "componentId": "sc-cIShpX",
        "isStatic": false,
        "rules": Array [
          "
  display: flex;
  flex-direction: column;

  ",
          [Function],
          ";
",
        ],
      },
      "displayName": "styled.div",
      "render": [Function],
      "styledComponentId": "sc-cIShpX",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [],
      },
      "ref": null,
      "rendered": Array [],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [
          Object {
            "data-test-id": [Function],
          },
        ],
        "componentStyle": ComponentStyle {
          "componentId": "sc-cIShpX",
          "isStatic": false,
          "rules": Array [
            "
  display: flex;
  flex-direction: column;

  ",
            [Function],
            ";
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-cIShpX",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
